language/move-lang/src/expansion/ast.rs

```diff
@@ -96,7 +96,14 @@ pub struct Function {
 // Specification Blocks
 //**************************************************************************************************
 
-// Specification block:
+#[derive(Debug, PartialEq)]
+pub struct FunctionPattern_ {
+    pub name_pattern: Vec<NamePatternFragment>,
+    pub type_arguments: Option<Vec<Type>>,
+}
+
```

Comment:
I'm really confused what these are used for. Do you have an example?

Reply:
The basic rationale is to avoid repeating specifications that apply to many functions in a module.

Reply:
I don't really know how this is used, or what the syntax is. Thats my confusion

Reply:
First, this feature is still experimental. But what ought to give us something like you find in aspect-oriented specification. We want to inject a uniform property given by a spec schema into all functions of a module which satisfy a certain pattern. Say

```
spec Module {
    apply Schema to withdraw*;  // all functions whose name starts with withdraw
    apply OtherSchema to public *;  // all functions which are public  
    apply GenericProperty<X> to *<X>; // all functions which we can instantiate with <X>
}
```
Notation is certainly subject of refinement.

Reply:
Ah, definitely seems like an interesting and powerful feature. I'll review the rest of this now that I understand whats going on :) Thanks!